eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/microfragments/eventdetails/fragments/views/EventDetailsView.java

```diff
@@ -76,16 +84,26 @@ public void update(EnrichedEvent enrichedEvent)
 
         new EventDescriptionView(mViews.schedjoulesEventDetailsDescription).update(new NullSafe<>(event.description()));
 
-        BaseActionFactory actionFactory = new BaseActionFactory();
-
-        new ActionView(mViews.schedjoulesEventDetailsActionShare)
-                .update(new OptionalAction(ApiLink.Rel.Action.SHARE, actionLinks, actionFactory, event));
-
-        new ActionView(mViews.schedjoulesEventDetailsActionAddToCalendar)
-                .update(new OptionalAction(ApiLink.Rel.Action.ADD_TO_CALENDAR, actionLinks, actionFactory, event));
+        final BaseActionFactory actionFactory = new BaseActionFactory();
 
+        LayoutInflater layoutInflater = LayoutInflater.from(mFragment.getContext());
+        for (Action action : new Flattened<>(
+                new Mapped<>(
+                        new ActionLinks(actionLinks, ApiLink.Rel.Action.BROWSE),
+                        new ActionFunction(actionFactory, event)),
+                new PresentValues<>(new ArrayIterable<Optional<Action>>(
+                        new OptionalAction(ApiLink.Rel.Action.SHARE, actionLinks, actionFactory, event),
+                        new OptionalAction(ApiLink.Rel.Action.ADD_TO_CALENDAR, actionLinks, actionFactory, event)))))
+        {
+            View actionView = layoutInflater.inflate(R.layout.schedjoules_view_event_details_action, mViews.schedjoulesEventDetailsActionContainer, false);
+            new ActionView(actionView, (TextView) actionView.findViewById(R.id.schedjoules_event_details_action_label)).update(action);
+            mViews.schedjoulesEventDetailsActionContainer.addView(actionView);
```

Comment:
Wasn't it the plan to have fix, hardcoded places for each of the actions and not add them dynamically?

Also considering https://github.com/schedjoules/android-event-discovery-sdk/issues/254 wouldn't it be simpler and more future proof to just add this new browse action the same way as it was with the others?

So just adding this to the layout:
```
<include
                            layout="@layout/schedjoules_view_event_details_browse"
                            android:id="@+id/schedjoules_event_details_action_browse"/>
```
And this to the code:
```
        new ActionView(mViews.schedjoulesEventDetailsActionBrowse)
                .update(new OptionalAction(ApiLink.Rel.Action.BROWSE, actionLinks, actionFactory, event));
```
So why the big refactor here creating this loop? Aren't we adding more complexity, and more work for later with this?
The previous state where each action is handled individually seemed more straightforward for me considering the UI plan to have fix places for them, which I suppose could even change, I mean that they could be displayed differently.

Reply:
Note that the place for each action is still hard coded and not determined by the API (which was the initial problem).
Anyway I made this change mostly because there can be multiple browse actions per event. If an event has multiple sources (e.g. Facebook + Meetup), there might be multiple source links. So I needed a loop anyway. Since "Share" and "Add To Calendar" are created the same way, it felt natural to include these in the loop.

Reply:
I see, I didn't know there can be more than one 'browse' action, I understand the change now.